requery-test/src/test/java/io/requery/test/AbstractObservableCollectionTest.java

```diff
@@ -0,0 +1,92 @@
+package io.requery.test;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+import io.requery.proxy.CollectionChanges;
+import io.requery.test.model.Phone;
+import io.requery.util.ObservableCollection;
+
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Created by mluchi on 25/05/2017.
+ */
+
+public abstract class AbstractObservableCollectionTest<T extends Collection<Phone> & ObservableCollection<Phone>> {
```

Comment:
This can be changed to a ParameterizedTest, that way you don't need to subclass it